ta/pkcs11/src/processing_symm.c

```diff
@@ -835,6 +837,43 @@ enum pkcs11_rc step_symm_operation(struct pkcs11_session *session,
 		}
 		break;
 
+	case PKCS11_CKM_AES_CBC_PAD:
+		if (step == PKCS11_FUNC_STEP_FINAL ||
+		    step == PKCS11_FUNC_STEP_ONESHOT)
+			break;
+
+		if (!in_buf) {
+			EMSG("No input data");
+			return PKCS11_CKR_ARGUMENTS_BAD;
+		}
+
+		switch (function) {
+		case PKCS11_FUNCTION_ENCRYPT:
+			res = TEE_CipherUpdate(proc->tee_op_handle,
+					       in_buf, in_size,
+					       out_buf, &out_size);
+
+			output_data = true;
+			rc = tee2pkcs_error(res);
+
+			/* Update text size still in the os buffer. */
+			if (rc == PKCS11_CKR_OK)
+				proc->last_block_size = (in_size + proc->last_block_size) & 0xF;
+
+			break;
+		case PKCS11_FUNCTION_DECRYPT:
```

Comment:
Prefer not introduce this if AES_PAD decryption is not supported

Reply:
> Hi @cmcgcc, thanks for the proposal. Since there's no decryption yet, i'm not sure this change will be merged for 3.17.0 tag. If so, are you ok we merge #5154 for the tag and add AES_PAD afterward?

Sorry for the late reply. I made some change to address the comments. I am working on the decryption part and will patch later.

Reply:
hi @cmcgcc. thanks for the update. you can close/hide these comment thread now if you want.